modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs

```diff
@@ -417,76 +417,31 @@ private static string CamelcaseToUnderscore(this string instance, bool lowerCase
+        /// Performs a comparison to another string and returns an integer that indicates their relative position in the sort order.
         /// </summary>
         /// <param name="instance">The string to compare.</param>
         /// <param name="to">The other string to compare.</param>
         /// <param name="caseSensitive">
         /// If <see langword="true"/>, the comparison will be case sensitive.
         /// </param>
-        /// <returns>-1 if less, 0 if equal and +1 if greater.</returns>
+        /// <returns>An integer that indicates the lexical relationship between the two comparands.</returns>
+        [Obsolete("Use string.Compare instead.")]
         public static int CompareTo(this string instance, string to, bool caseSensitive = true)
         {
-            if (string.IsNullOrEmpty(instance))
-                return string.IsNullOrEmpty(to) ? 0 : -1;
-
-            if (string.IsNullOrEmpty(to))
-                return 1;
-
-            int instanceIndex = 0;
-            int toIndex = 0;
-
-            if (caseSensitive) // Outside while loop to avoid checking multiple times, despite some code duplication.
-            {
-                while (true)
-                {
-                    if (to[toIndex] == 0 && instance[instanceIndex] == 0)
-                        return 0; // We're equal
-                    if (instance[instanceIndex] == 0)
-                        return -1; // If this is empty, and the other one is not, then we're less... I think?
-                    if (to[toIndex] == 0)
-                        return 1; // Otherwise the other one is smaller...
-                    if (instance[instanceIndex] < to[toIndex]) // More than
-                        return -1;
-                    if (instance[instanceIndex] > to[toIndex]) // Less than
-                        return 1;
-
-                    instanceIndex++;
-                    toIndex++;
-                }
-            }
-            else
-            {
-                while (true)
-                {
-                    if (to[toIndex] == 0 && instance[instanceIndex] == 0)
-                        return 0; // We're equal
-                    if (instance[instanceIndex] == 0)
-                        return -1; // If this is empty, and the other one is not, then we're less... I think?
-                    if (to[toIndex] == 0)
-                        return 1; // Otherwise the other one is smaller..
-                    if (char.ToUpperInvariant(instance[instanceIndex]) < char.ToUpperInvariant(to[toIndex])) // More than
-                        return -1;
-                    if (char.ToUpperInvariant(instance[instanceIndex]) > char.ToUpperInvariant(to[toIndex])) // Less than
-                        return 1;
-
-                    instanceIndex++;
-                    toIndex++;
-                }
-            }
+            return string.Compare(instance, to, !caseSensitive);
```

Comment:
@paulloz Are you OK with using this overload in the new implementation? It will use the current culture: https://github.com/dotnet/runtime/blob/ee93104aa0e7ac77ca7e4c6db9873356a9e3e622/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs#L210

Reply:
Yep, I think it's fine. Anyone is then free to use another locale if they wish anyway :+1:

Reply:
I was not thinking, but calling `string.Compare(string, string, bool)` (instead of `string.Compare(string, string, bool, CultureInfo)`) warns with `CA1304` and `CA1309`.